src/new-client/src/app/lib/types/index.tsx

```diff
@@ -126,8 +126,103 @@ type Session = {
   services: string[];
 };
 
+type FetchAvailabilityRequest = {
+  sites: string[];
+  service: string;
+  from: string;
+  until: string;
+  queryType: string;
+};
+
+type AvailabilityResponse = {
+  site: string;
+  service: string;
+  availability: Availability[];
+};
+
+type Availability = {
+  date: Date;
+  blocks: AvailabilityBlock[];
+};
+
+type AvailabilityBlock = {
+  from: string;
+  until: string;
+  count: number;
+};
+
+// TODO: Rename this type to something more meaningful
+// TODO: Do we need to include start year & end year?
+type Week = {
+  start: number;
+  end: number;
+  startMonth: number;
+  endMonth: number;
+  startYear: number;
+  endYear: number;
+  unbooked?: number;
+  totalAppointments?: number;
+  booked?: number;
+  bookedAppointments: bookedAppointments[];
+};
+
+type FetchBookingsRequest = {
+  from: string;
+  to: string;
+  site: string;
+};
+
+type Booking = {
+  reference: string;
+  from: string;
+  duration: number;
+  service: string;
+  site: string;
+  outcome?: string;
+  attendeeDetails: AttendeeDetails;
+  contactDetails?: ContactItem[];
+  reminderSet: boolean;
+  created: string;
+  provisional: boolean;
```

Comment:
@vicr1 has just changed this property on the backend to be a new type object describing booked/provisional as a status, so might need updating here